batch/batch/front_end/templates/job.html

```diff
@@ -1,73 +1,141 @@
+  </thead>
+  <tbody>
+    {% for attempt in attempts %}
+    <tr>
+	    <td class="numeric-cell">{{ attempt['attempt_id'] }}</td>
+	    <td>{{ attempt['instance_name'] }}</td>
+	    <td>
+	      {% if 'start_time' in attempt and attempt['start_time'] is not none %}
+	      {{ attempt['start_time'] }}
+	      {% endif %}
+	    </td>
+	    <td>
+	      {% if 'end_time' in attempt and attempt['end_time'] is not none %}
+	      {{ attempt['end_time'] }}
+	      {% endif %}
+	    </td>
+	    <td>
+	      {% if 'duration' in attempt and attempt['duration'] is not none %}
+	      {{ attempt['duration'] }}
+	      {% endif %}
+	    </td>
+	    <td>
+	      {% if 'reason' in attempt and attempt['reason'] is not none %}
+	      {{ attempt['reason'] }}
+	      {% endif %}
+	    </td>
+    </tr>
+    {% endfor %}
+  </tbody>
+</table>
+{% else %}
+<p>No attempts</p>
+{% endif %}
 
-    {% if 'input' in job_log %}
-      <h3>Input</h3>
-      <pre>{{ job_log['input'] }}</pre>
+<h2>Step Status</h2>
+<table class="data-table">
+  <thead>
+    <tr>
+	    <th>Job Step</th>
+	    <th>Image Pulling Time (s)</th>
+	    <th>Running Time (s)</th>
+	    <th>Exceeded Memory Limits?</th>
+	    <th>State</th>
+    </tr>
+  </thead>
+  <tbody>
+    {% for step in step_statuses %}
+    {% if step %}
+    <tr>
+	    <td>{{ step['name'] }}</td>
+	    <td>
+        {% if step['timing']['pulling'] %}
+        {{ step['timing']['pulling']['duration'] / 1000.0 }}
+        {% endif %}
+      </td>
+	    <td>
+        {% if step['timing']['running'] %}
+        {{ step['timing']['running']['duration'] / 1000.0 }}
+        {% endif %}
+      </td>
+	    {% if step['container_status']['out_of_memory'] %}
+      <td class="data-table-bad" >Yes</td>
+      {% else %}
+      <td>No</td>
+      {% endif %}
+      <td {% if step['state'] in ['failed', 'error'] %}class="data-table-bad"{% endif %}>
+        {{ step['state'] }}
+      </td>
+    </tr>
     {% endif %}
+    {% endfor %}
+  </tbody>
+</table>
 
-    {% if 'main' in job_log %}
-      <h3>Main</h3>
-      <pre>{{ job_log['main'] }}</pre>
-    {% endif %}
+{% if job_log %}
+<h2>Log</h2>
 
-    {% if 'output' in job_log %}
-      <h3>Output</h3>
-      <pre>{{ job_log['output'] }}</pre>
-    {% endif %}
+{% if 'input' in job_log %}
+<h3>Input</h3>
+<pre>{{ job_log['input'] }}</pre>
+{% endif %}
+
+{% if 'main' in job_log %}
+<h3>Main</h3>
+<pre>{{ job_log['main'] }}</pre>
+{% endif %}
+
+{% if 'output' in job_log %}
+<h3>Output</h3>
+<pre>{{ job_log['output'] }}</pre>
+{% endif %}
+{% endif %}
+
+<h2>Job Specification</h2>
+<table class="data-table">
+  <thead>
+    <tr>
+	    <th>Name</th>
+	    <th>Value</th>
+    </tr>
+  </thead>
+  {% if job_specification %}
+  <tbody>
+    <tr>
+      <td>image</td>
+      <td>{{ job_specification['image'] }}</td>
+    </tr>
+    <tr>
+      <td>command</td>
+      <td><pre style="max-height: 10em; max-width: 80em;">{{ "'" + (job_specification['command'] | join("' '")) + "'" }}</pre></td>
```

Comment:
Do we need environment variables as well for the command to be helpful to the user?

Reply:
<img width="654" alt="Screen Shot 2020-12-03 at 2 16 51 PM" src="https://user-images.githubusercontent.com/106194/101077045-3ebc3e00-3572-11eb-8be4-324d2fac9654.png">